File: Nestings.js
function Nesting(S){
  let brackets = {
    "(": ")"
  }
  let stack = []

  for( let i = 0; i < S.length; i++ ){
    if ( brackets[S.charAt(i)] ) {
      stack.push(S.charAt(i))
    } else if ( brackets[stack[stack.length-1]] === S.charAt(i) ){
      stack.pop()
    } else {
      return 0
    }
  }

  if (stack.length === 0){
    return 1
  } else {
    return 0
  }
}

//console.log(Nesting("((()))"))
//console.log(Nesting("(())("))
//console.log(Nesting("(()))"))

File: py/nestings.py
def solution(S):
    # write your code in Python 3.6
    brackets = {
        '(' : ')'
    }
    stack = []
    
    for i in range(len(S)):
        if S[i] in brackets.keys():
            stack.append(S[i])
        elif len(stack) > 0 and brackets[stack[-1]] == S[i]:
            stack.pop()
        else:
            return 0
        
    if (len(stack) == 0):
        return 1
    else:
        return 0
